src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -591,7 +591,10 @@ public override BoundNode VisitNameOfOperator(BoundNameOfOperator node)
 
         public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperator node)
         {
-            if (_inExpressionLambda && (node.LeftOperand.IsLiteralNull() || node.LeftOperand.IsLiteralDefault()))
+            if (_inExpressionLambda &&
+                (node.LeftOperand.IsLiteralNull() ||
+                 node.LeftOperand.IsLiteralDefault() ||
+                 node.LeftOperand.IsTypelessNew()))
```

Comment:
`null ?? T` is allowed but `new() ?? e` and `default ?? e` are not (already an error). Should I revert both here?

Reply:
I didn't quite follow your question. I would think that `new() ?? e` should be disallowed in code and expression tree.

---
In reply to: [209343381](https://github.com/dotnet/roslyn/pull/29167#discussion_r209343381) [](ancestors = 209343381)

Reply:
I'm saying that `IsLiteralDefault` is added here by mistake. because `default ?? e` is already an error. same for `new()` - I added the check here following `IsLiteralDefault` but then realized both are redundant.

The only exception is `null` which isn't allowed in expression trees but it's valid anywhere else.

Reply:
Thanks for the clarification. It's fine to remove the checks for `IsLiteralDefault` and `IsTypelessNew` since already handled somewhere else.